WordPress/Classes/Services/MediaCoordinator.swift

```diff
@@ -543,42 +543,34 @@ class MediaCoordinator: NSObject {
     }
 
     /// Encapsulates an observer block and an optional observed media item or post.
-    struct MediaObserver {
-        let media: Media?
-        let post: AbstractPost?
+    private struct MediaObserver {
+        let mediaObjectID: NSManagedObjectID?
+        let postObjectID: NSManagedObjectID?
         let onUpdate: ObserverBlock
 
-        init(onUpdate: @escaping ObserverBlock) {
-            self.media = nil
-            self.post = nil
-            self.onUpdate = onUpdate
-        }
-
         init(media: Media?, onUpdate: @escaping ObserverBlock) {
-            self.media = media
-            self.post = nil
+            self.mediaObjectID = media?.objectID
+            self.postObjectID = nil
             self.onUpdate = onUpdate
         }
 
         init(post: AbstractPost, onUpdate: @escaping ObserverBlock) {
-            self.media = nil
-            self.post = post
+            self.mediaObjectID = nil
+            self.postObjectID = post.objectID
             self.onUpdate = onUpdate
         }
     }
 
     /// Utility method to return all observers for a specific media item,
     /// including any 'wildcard' observers that are observing _all_ media items.
```

Comment:
This comment is out of date now 🤔 Could we update it to

```suggestion
    /// Utility method to return all observers for a `Media` item with the given `NSManagedObjectID`
    /// and part of the posts with given `NSManagedObjectID`s, including any 'wildcard' observers 
    /// that are observing _all_ media items.
```